app/webpacker/components/ContactEditProfilePage/index.jsx

```diff
@@ -8,21 +8,37 @@ import Loading from '../Requests/Loading';
 import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
 import Errored from '../Requests/Errored';
 import EditProfileForm from './EditProfileForm';
+import useLoggedInUserPermissions from '../../lib/hooks/useLoggedInUserPermissions';
+import useQueryParams from '../../lib/hooks/useQueryParams';
+import useInputState from '../../lib/hooks/useInputState';
+import { IdWcaSearch } from '../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../SearchWidget/SearchModel';
 
 const CONTACT_EDIT_PROFILE_QUERY_CLIENT = new QueryClient();
 
 export default function ContactEditProfilePage({ loggedInUserId, recaptchaPublicKey }) {
+  const [queryParams] = useQueryParams();
+  const editOthersProfileMode = Boolean(queryParams.editOthersProfile);
   const { data: loggedInUserData, isLoading, isError } = useQuery({
     queryKey: ['userData'],
     queryFn: () => fetchJsonOrError(apiV0Urls.users.me.userDetails),
-    enabled: !!loggedInUserId,
+    enabled: (
+      // If not logged in, fetching WCA ID of logged in user is not possible.
+      !!loggedInUserId
+      // If the user needs to edit other's profile, then fetching own WCA ID is not needed.
+       || !editOthersProfileMode
```

Comment:
It took me a while to see thrrough this. I think a better explanation of the `!` should be: If the user is *not* editing somebody else's profile, then we need to fetch their own WCA ID.

Reply:
(The problem is that your comment explains when the fetching should be DISabled, but the `useQuery` function needs to be passed information when it is ENabled. So I was confused why your comment says "if the user needs to edit other's profile", while the code directly below says "NOT editing other's profile")